source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -44,23 +44,35 @@ void Filter::initiateCall(const Http::RequestHeaderMap& headers) {
   envoy::config::core::v3::Metadata metadata_context;
 
   // If metadata_context_namespaces is specified, pass matching filter metadata to the ext_authz
-  // service.
+  // service. If metadata key is set in both the connection and request metadata then the value
+  // will be the request metadata value.
+  const auto& connection_metadata =
+      decoder_callbacks_->connection()->streamInfo().dynamicMetadata().filter_metadata();
   const auto& request_metadata =
       decoder_callbacks_->streamInfo().dynamicMetadata().filter_metadata();
   for (const auto& context_key : config_->metadataContextNamespaces()) {
-    if (const auto& metadata_it = request_metadata.find(context_key);
-        metadata_it != request_metadata.end()) {
+    if (const auto& metadata_it = connection_metadata.find(context_key);
+        metadata_it != connection_metadata.end()) {
+      (*metadata_context.mutable_filter_metadata())[metadata_it->first] = metadata_it->second;
+    } else if (const auto& metadata_it = request_metadata.find(context_key);
+               metadata_it != request_metadata.end()) {
```

Comment:
I think we should try find request_metadata first because it has higher priority? Please add a test to ensure the request metadata will  be taken as priority if both are exist, Thanks.

And by the way, value semantic should be used for iterator, so `const auto metadata_it = ...` may be better here.

Reply:
You are right, the order is incorrect. This is why the presubmit tests are failing. I added the unit tests at: https://github.com/envoyproxy/envoy/pull/27612/files#diff-64a51bc8f4c7ce4f31a3420225e6fd4d0f14a187424148da4b9dc87e11d97c96R1426-R1432